src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -614,142 +615,197 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
+      # malformed expressions, and cause scalar conversion below to fail)
+      names <- vapply(opts, is.name, TRUE)
+      opts[names] <- as.character(opts[names])
+      
+   })
+   
+   # now, try to find inline YAML chunk options
+   yamlChunkOpts <- c()
+   
+   # figure out the appropriate chunk prefix based on the engine
+   engine <- tolower(.rs.nullCoalesce(opts$engine, "r"))
+   commentPrefix <- if (engine %in% c("cc", "cpp", "go", "js", "node", "rcpp", "stan"))
+      "/{2,}"
+   else if (engine %in% c("haskell", "sql"))
+      "[-]{2,}"
+   else
+      "#+"
+   
+   yamlPrefix <- .rs.reYamlOptChunkBegin(commentPrefix)
+   for (line in code[-1]) {
+      
+      # there may be variable number of yaml chunk opts, but they should all be at the top of the chunk
+      match <- unlist(regmatches(line, regexec(yamlPrefix, line)))
+      if (length(match) == 0)
+         break
+      
+      # add the line
+      yamlChunkOpts <- c(yamlChunkOpts, match)
+      
+   }
+   
+   # if we had some YAML chunk options, try to add them
+   if (length(yamlChunkOpts) > 0)
+   {
+      tryCatch({
+         yamlOpts <- .rs.parseYamlOpt(yamlChunkOpts, commentPrefix)
+         opts <- .rs.mergeLists(opts, yamlOpts)
+      }, error = function(e) {
+         warning("Failed to parse chunk options in body:\n", e)
+      })
+   }
+   
+   # ensure that fields we expect to be logical are actually logical
+   # (sanitize invalid inputs here since they can break notebook execution)
+   fields <- list(
       warning = TRUE,
       message = TRUE,
       error   = FALSE
-    )
-    
-    .rs.enumerate(fields, function(key, default) {
-       
+   )
+   
+   .rs.enumerate(fields, function(key, default) {
+      
       if (is.null(opts[[key]]))
-        next
-       
+         return()
+      
       opts[[key]] <<- tryCatch(
-        as.logical(opts[[key]]),
-        error = function(e) default
+         as.logical(opts[[key]]),
+         error = function(e) default
       )
-       
-    })
-    
-  },
-  error = function(e) {})
-
-  .rs.scalarListFromList(opts, expressions = TRUE)
+      
+   })
+   
+   # if this is a SQL chunk, try to provide a default connection
+   # TODO: arguably, we should be using `knitr::opts_chunk$get()`
+   # to provide defaults for chunk options, but it's not clear if
+   # that's always the correct choice, so we just make the smaller
+   # piece-meal decision here
+   if (opts$engine %in% c("mysql", "psql", "sql")) {
+      
+      # get current connection
+      conn <- .rs.nullCoalesce(
+         opts$connection,
+         knitr::opts_chunk$get("connection")
+      )
+      
+      # our chunk execution tools expect a character connection,
+      # so use a temporary variable to store the connection value
+      if (!is.null(conn) && !is.character(conn)) {
+         .rs.setVar("chunkConnection", conn)
+         opts$connection <- ".rs.chunkConnection"
+      }
```

Comment:
These changes were kind of hard to find, especially amidst all the whitespace changes -- this probably could have been its own PR, but thanks for fixing this

Reply:
Sorry about that -- since I was in this part of the code, I wanted to dig through other related pieces.